File: MMORPG_2/Assets/Script/Data/LocalData/CreateLua/SkillEntity.lua
SkillEntity = { Id = 0, SkillName = "", SkillDesc = "", EffectName = "", SkillIconFolder = "", SkillIconName = "", HurtValueRate = 0, HurtValueRateUp = 0, LevelLimit = 0, SpendMP = 0, SpendMPLevelUp = 0, IsPhyAttack = 0, IsDoCameraShake = 0, CameraShakeDelay = 0, SkillCD = 0, SkillUpCDTime = 0, EffectLifeTime = 0, AttackTargetCount = 0, IsTargetMonster = 0, AttackRange = 0, AreaAttackRadius = 0, ShowEffectDelay = 0, ShowHurtEffectDelaySecond = 0, RedScreen = 0, AttackState = 0, AbnormalState = 0, BuffInfoID = 0, BuffTargetFilter = 0, BuffIsPercentage = 0 }

--这句是重定义元表的索引，就是说有了这句，这个才是一个类
SkillEntity.__index = SkillEntity;

function SkillEntity.New(Id, SkillName, SkillDesc, EffectName, SkillIconFolder, SkillIconName, HurtValueRate, HurtValueRateUp, LevelLimit, SpendMP, SpendMPLevelUp, IsPhyAttack, IsDoCameraShake, CameraShakeDelay, SkillCD, SkillUpCDTime, EffectLifeTime, AttackTargetCount, IsTargetMonster, AttackRange, AreaAttackRadius, ShowEffectDelay, ShowHurtEffectDelaySecond, RedScreen, AttackState, AbnormalState, BuffInfoID, BuffTargetFilter, BuffIsPercentage)
    local self = { }; --初始化self
    setmetatable(self, SkillEntity); --将self的元表设定为Class

    self.Id = Id;
    self.SkillName = SkillName;
    self.SkillDesc = SkillDesc;
    self.EffectName = EffectName;
    self.SkillIconFolder = SkillIconFolder;
    self.SkillIconName = SkillIconName;
    self.HurtValueRate = HurtValueRate;
    self.HurtValueRateUp = HurtValueRateUp;
    self.LevelLimit = LevelLimit;
    self.SpendMP = SpendMP;
    self.SpendMPLevelUp = SpendMPLevelUp;
    self.IsPhyAttack = IsPhyAttack;
    self.IsDoCameraShake = IsDoCameraShake;
    self.CameraShakeDelay = CameraShakeDelay;
    self.SkillCD = SkillCD;
    self.SkillUpCDTime = SkillUpCDTime;
    self.EffectLifeTime = EffectLifeTime;
    self.AttackTargetCount = AttackTargetCount;
    self.IsTargetMonster = IsTargetMonster;
    self.AttackRange = AttackRange;
    self.AreaAttackRadius = AreaAttackRadius;
    self.ShowEffectDelay = ShowEffectDelay;
    self.ShowHurtEffectDelaySecond = ShowHurtEffectDelaySecond;
    self.RedScreen = RedScreen;
    self.AttackState = AttackState;
    self.AbnormalState = AbnormalState;
    self.BuffInfoID = BuffInfoID;
    self.BuffTargetFilter = BuffTargetFilter;
    self.BuffIsPercentage = BuffIsPercentage;

    return self;
end

File: MMORPG_2/Assets/Script/Data/LocalData/Create/SkillEntity.cs

//===================================================
//作    者：
//创建时间：2019-01-17 15:18:26
//备    注：此代码为工具生成 请勿手工修改
//===================================================
using System.Collections;

/// <summary>
/// Skill实体
/// </summary>
public partial class SkillEntity : AbstractEntity
{
    /// <summary>
    /// 技能名称
    /// </summary>
    public string SkillName { get; set; }

    /// <summary>
    /// 技能描述
    /// </summary>
    public string SkillDesc { get; set; }

    /// <summary>
    /// 特效名
    /// </summary>
    public string EffectName { get; set; }

    /// <summary>
    /// 技能图片所在图集
    /// </summary>
    public string SkillIconFolder { get; set; }

    /// <summary>
    /// 技能图片名
    /// </summary>
    public string SkillIconName { get; set; }

    /// <summary>
    /// 技能伤害倍率
    /// </summary>
    public int HurtValueRate { get; set; }

    /// <summary>
    /// 技能每升一级提升的倍率
    /// </summary>
    public int HurtValueRateUp { get; set; }

    /// <summary>
    /// 技能最大等级
    /// </summary>
    public int LevelLimit { get; set; }

    /// <summary>
    /// 施放技能所需MP
    /// </summary>
    public int SpendMP { get; set; }

    /// <summary>
    /// 技能每升一级所需要增加的MP
    /// </summary>
    public int SpendMPLevelUp { get; set; }

    /// <summary>
    /// 是否物理攻击
    /// </summary>
    public int IsPhyAttack { get; set; }

    /// <summary>
    /// 是否震屏
    /// </summary>
    public int IsDoCameraShake { get; set; }

    /// <summary>
    /// 震屏延迟时间
    /// </summary>
    public float CameraShakeDelay { get; set; }

    /// <summary>
    /// 技能冷却时间(秒)
    /// </summary>
    public float SkillCD { get; set; }

    /// <summary>
    /// 技能每升一级减多少冷却时间(秒)
    /// </summary>
    public float SkillUpCDTime { get; set; }

    /// <summary>
    /// 特效生命时长
    /// </summary>
    public float EffectLifeTime { get; set; }

    /// <summary>
    /// 伤害目标数量
    /// </summary>
    public int AttackTargetCount { get; set; }

    /// <summary>
    /// 技能是否对怪释放
    /// </summary>
    public int IsTargetMonster { get; set; }

    /// <summary>
    /// 此技能攻击攻击范围(米)
    /// </summary>
    public float AttackRange { get; set; }

    /// <summary>
    /// 群攻的伤害判定半径
    /// </summary>
    public float AreaAttackRadius { get; set; }

    /// <summary>
    /// 动作播放多久后才播放特效
    /// </summary>
    public float ShowEffectDelay { get; set; }

    /// <summary>
    /// 攻击动作发出多少秒后被攻击者才播放受伤效果
    /// </summary>
    public float ShowHurtEffectDelaySecond { get; set; }

    /// <summary>
    /// 主角被这个物理子攻击后，是否会导致屏幕四周泛红
    /// </summary>
    public int RedScreen { get; set; }

    /// <summary>
    /// 攻击后状态
    /// </summary>
    public int AttackState { get; set; }

    /// <summary>
    /// 附加异常状态
    /// </summary>
    public int AbnormalState { get; set; }

    /// <summary>
    /// BuffInfoID
    /// </summary>
    public int BuffInfoID { get; set; }

    /// <summary>
    /// Buff类型
    /// </summary>
    public int BuffTargetFilter { get; set; }

    /// <summary>
    /// Buff效果值是否按百分比计算
    /// </summary>
    public int BuffIsPercentage { get; set; }

}
